frontend/src/components/pages/shadowlinks/details/shadowlink-details-page.tsx

```diff
@@ -0,0 +1,246 @@
+/**
+ * Copyright 2025 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file https://github.com/redpanda-data/redpanda/blob/dev/licenses/bsl.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+'use client';
+
+import { Button } from 'components/redpanda-ui/components/button';
+import { Text } from 'components/redpanda-ui/components/typography';
+import { AlertCircle, Loader2, Trash2 } from 'lucide-react';
+import { runInAction } from 'mobx';
+import { useEffect, useMemo, useState } from 'react';
+import {
+  useDeleteShadowLinkMutation,
+  useFailoverShadowLinkMutation,
+  useGetShadowLinkQuery,
+  useListShadowTopicInfiniteQuery,
+} from 'react-query/api/shadowlink';
+import { useNavigate, useParams } from 'react-router-dom';
+import { toast } from 'sonner';
+import { uiState } from 'state/ui-state';
+
+import { DeleteShadowLinkDialog } from './delete-shadowlink-dialog';
+import { FailoverDialog } from './failover-dialog';
+import { ShadowLinkDiagram } from './shadow-link-diagram';
+import { ShadowLinkMetrics } from './shadow-link-metrics';
+import { ShadowTopicsTable } from './shadow-topics-table';
+
+// Update page title using uiState pattern
+export const updatePageTitle = (shadowLinkName: string) => {
+  runInAction(() => {
+    uiState.pageTitle = shadowLinkName;
+    uiState.pageBreadcrumbs = [
+      { title: 'Shadow Links', linkTo: '/shadowlinks' },
+      { title: shadowLinkName, linkTo: `/shadowlinks/${shadowLinkName}` },
+    ];
+  });
+};
+
+export const ShadowLinkDetailsPage = () => {
+  const { name } = useParams<{ name: string }>();
+  const navigate = useNavigate();
+  const [showDeleteDialog, setShowDeleteDialog] = useState(false);
+  const [showFailoverDialog, setShowFailoverDialog] = useState(false);
+  const [failoverTopicName, setFailoverTopicName] = useState<string>('');
+  const [topicNameFilter, setTopicNameFilter] = useState('');
+
+  // Update page title
+  useEffect(() => {
+    if (name) {
+      updatePageTitle(name);
+    }
+  }, [name]);
+
+  // Fetch shadow links data
+  const { data: shadowLinksData, isLoading, error: errorGetShadowLink, refetch } = useGetShadowLinkQuery({ name });
+  const {
+    data: shadowTopicsData,
+    refetch: refetchTopics,
+    fetchNextPage,
+    hasNextPage,
+    isFetchingNextPage,
+    isFetching: isFetchingTopics,
+  } = useListShadowTopicInfiniteQuery(
+    {
+      shadowLinkName: name,
+      filter: topicNameFilter
+        ? {
+            topicNameContains: topicNameFilter,
+          }
+        : undefined,
+      pageSize: 100,
```

Comment:
Let's use `MAX_PAGE_SIZE` import

Reply:
Sure, could you tell me why our max page is 500?